infrastructure/sentry-subgraph/src/utils/updateChallenge.ts

```diff
@@ -39,15 +39,76 @@ export function updateChallenge(referee: Referee, challenge: Challenge): Challen
     challenge.numberOfEligibleClaimers = challengeStruct.numberOfEligibleClaimers;
     challenge.amountClaimedByClaimers = challengeStruct.amountClaimedByClaimers;
 
-    //set challenge field in NodeConfirmed event to challenge id and save
-    let nodeConfirmation = NodeConfirmation.load(challenge.assertionId.toString());
-    if (!nodeConfirmation) {
-        log.warning(`Failed to load nodeConfirmation entity with id: ${challenge.assertionId}`, []);
+    //return early here because assertion checking was not turned on until this block number
+    let firstAssertionCheckingBlockNumber = 176156007;
+    if (event.block.number.lt(BigInt.fromI32(firstAssertionCheckingBlockNumber))) {
         return challenge;
     }
-    nodeConfirmation.challenge = challenge.id.toString();
-    nodeConfirmation.save();
+
+    //set challenge id range and load previous challenge entity
+    const currentChallengeId = challenge.challengeNumber;
+    const previousChallengeId = currentChallengeId.minus(BigInt.fromI32(1));
+    const previousChallenge = Challenge.load(previousChallengeId.toString());
+
+    //previous challenge not found, therefore on the first challenge
+    if (!previousChallenge) {
+        //set challenge field in NodeConfirmed event to challenge id and save
+        let nodeConfirmation = NodeConfirmation.load(challenge.assertionId.toString());
+        if (!nodeConfirmation) {
+            log.warning(`Failed to load nodeConfirmation entity with id: ${challenge.assertionId}`, []);
+            return challenge;
+        }
+        nodeConfirmation.challenge = challenge.id;
+        nodeConfirmation.save();
+        return challenge;
+    }
```

Comment:
I think this whole block should not be needed as we return for all the challenges before `firstAssertionCheckingBlockNumber `, let me know if that makes sense.

Reply:
Its still needed as a null guard.